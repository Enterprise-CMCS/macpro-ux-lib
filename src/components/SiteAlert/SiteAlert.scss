@use "../../sass/macbis-colors" as c;

.usa-site-alert {
  button {
    border: none;
    background-color: transparent;
  }

  .usa-alert {
    border: none;
  }
}

.usa-site-alert--info,
.usa-site-alert--info .usa-alert {
  background-color: c.$primary-alt-lightest;

  .usa-alert__text {
    color: c.$primary-darker;
  }
}

.usa-site-alert--emergency {
  .usa-alert {
    background-color: c.$error-dark;

    .usa-alert__body .usa-link {
      color: white;
    }
  }
}

.usa-site-alert .usa-alert .usa-alert__body {
  background-color: transparent;
}
