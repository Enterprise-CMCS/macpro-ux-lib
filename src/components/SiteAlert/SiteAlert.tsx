import React from "react";

type IntrinsicElements = JSX.IntrinsicElements["div"];

interface Props extends IntrinsicElements {
  alertHeading?: string;
  alertBody: string | React.ReactNode;
  emergency?: boolean;
  slim?: boolean;
  icon?: boolean;
  close?: React.MouseEventHandler<HTMLButtonElement>;
}

/**
 * **SiteAlert Component**
 *
 * @param {string}                     alertHeading   Text content for the alert heading.
 * @param {string | React.ReactNode}   alertBody      Text content for the alert body.
 * @param {string}                     emergency      Determines the alert whether to render the information or emergency alert.
 * @param {boolean}                    slim           Renders a slimmer version of the banner.
 * @param {boolean}                    icon           Show/hide the alert icon associated with the variation of the alert.
 * @param {React.MouseEventHandler}    close          Close button that shows when a event handler is passed to the component.
 */
export const SiteAlert: React.FC<Props> = ({
  alertHeading,
  alertBody,
  emergency = false,
  slim = false,
  icon = true,
  close,
  role = "alert",
  ...rest
}) => {
  return (
    <section
      className={`usa-site-alert usa-site-alert--${
        emergency ? "emergency" : "info"
      }${slim ? " usa-site-alert--slim" : ""}${
        icon ? "" : " usa-site-alert--no-icon"
      }`}
      aria-label="Site alert,"
      {...rest}
    >
      <div className="usa-alert">
        <div className="usa-alert__body">
          {alertHeading && (
            <h3 className="usa-alert__heading">{alertHeading}</h3>
          )}
          <div className="usa-alert__text">{alertBody}</div>
        </div>
      </div>
    </section>
  );
};
