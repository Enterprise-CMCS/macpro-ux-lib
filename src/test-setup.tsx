import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
