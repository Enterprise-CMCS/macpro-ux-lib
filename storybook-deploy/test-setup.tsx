import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };

/**
 * Clean up attributes from the rendered component by removing the generated ids.
 *
 * Before:
 *   for="input-type-text-123456"
 *   id="input-type-text-123456"
 *
 * After:
 *   for="input-type-text"
 *   id="input-type-text"
 *
 * This is useful during snapshot testing. Snapshot comparisons will fail without addressing these items.
 *
 * @param {HTMLElement} container The component to clean up.
 * @param {prefixes}      string[]  The reference substrings to leave alone.
 */
