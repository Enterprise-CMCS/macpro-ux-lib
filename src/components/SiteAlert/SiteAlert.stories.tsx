import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { SiteAlert } from "./SiteAlert";

export default {
  title: "COMPONENTS/SiteAlert",
  component: SiteAlert,
  args: {
    alertHeading: "Emergency alert message",
    alertBody: [
      <div key="alert-3">
        <strong>Short alert message.</strong> Additional context and followup
        information including{" "}
        <a className="usa-link" href="https://google.com">
          a link
        </a>
        .
      </div>,
    ],
  },
  argTypes: {
    alertHeading: {
      description: "Text content for the alert heading.",
    },
    alertBody: {
      control: "text",
      description:
        "Text content for the alert body. Users can provide either a string or React Children to display for the alert body.",
    },
    emergency: {
      description:
        "Determines the alert whether to render the information or emergency alert.",
    },
    slim: {
      description: "Renders a slimmer version of the banner.",
    },
    icon: {
      description:
        "Show/hide the alert icon associated with the variation of the alert.",
    },
    close: {
      description:
        "Close button that shows when a event handler is passed to the component",
      control: false,
    },
  },
  parameters: {
    docs: {
      description: {
        component:
          "An alert keeps users informed of important and sometimes time-sensitive changes.",
      },
    },
  },
} as ComponentMeta<typeof SiteAlert>;

const Template: ComponentStory<typeof SiteAlert> = ({ ...rest }) => (
  <SiteAlert {...rest} />
);

export const Info = Template.bind({});
Info.args = {
  alertBody: [
    <p key="info-1" className="usa-alert__text">
      Additional context and followup information including{" "}
      <a className="usa-link" href="https://google.com">
        a link
      </a>
      .
    </p>,
  ],
};

export const InfoSiteAlertWithCloseButton = Template.bind({});
InfoSiteAlertWithCloseButton.args = {
  close: () => console.log("test"),
};

export const EmergencySiteAlert = Template.bind({});
EmergencySiteAlert.args = {
  emergency: true,
};

export const EmergencySiteAlertWithNoHeader = Template.bind({});
EmergencySiteAlertWithNoHeader.args = {
  alertHeading: "",
  emergency: true,
};

export const EmergencySiteAlertWithList = Template.bind({});
EmergencySiteAlertWithList.args = {
  emergency: true,
  alertBody: [
    <ul key="alert-2" className="usa-list">
      <li>
        The primary emergency message and{" "}
        <a className="usa-link" href="https://google.com">
          a link
        </a>{" "}
        for supporting context.
      </li>
      <li>
        Another message,{" "}
        <a className="usa-link" href="https://google.com">
          and another link
        </a>
        .
      </li>
      <li>A final emergency message.</li>
    </ul>,
  ],
};

export const SlimEmergencySiteAlert = Template.bind({});
SlimEmergencySiteAlert.args = {
  emergency: true,
  slim: true,
  alertHeading: "",
};

export const SlimSiteAlertEmergencyWithNoIcon = Template.bind({});
SlimSiteAlertEmergencyWithNoIcon.args = {
  emergency: true,
  icon: false,
  alertHeading: "",
};
