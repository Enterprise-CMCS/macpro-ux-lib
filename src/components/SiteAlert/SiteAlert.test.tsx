import React from "react";
import { fireEvent, screen, render } from "../../test-setup";
import { SiteAlert } from "./SiteAlert";

describe("Tests for the SiteAlert component", () => {
  it("should render an info alert", () => {
    render(
      <SiteAlert
        data-testid="SiteAlert"
        alertHeading="Test"
        alertBody="Test body"
      />
    );

    const AlertComp = screen.getByTestId("SiteAlert");

    expect(AlertComp).toBeInTheDocument();
    expect(AlertComp).toHaveTextContent("Test");
    expect(AlertComp).toHaveTextContent("Test body");
  });

  it("should render an emergency slim alert without an icon", () => {
    render(
      <SiteAlert
        data-testid="SiteAlert"
        alertHeading="Warning Heading"
        alertBody="Warning Body"
        emergency
        icon={false}
        slim
        close={() => {
          console.log("test");
        }}
      />
    );

    const AlertComp = screen.getByTestId("SiteAlert");

    expect(AlertComp).toHaveTextContent("Warning Heading");
    expect(AlertComp).toHaveTextContent("Warning Body");
    expect(AlertComp).toHaveClass("usa-site-alert--slim");
    expect(AlertComp).toHaveClass("usa-site-alert--no-icon");
  });
});

describe("compontent snapshots", () => {
  it("should render an emergency alert", () => {
    const { container } = render(
      <SiteAlert emergency alertHeading="Test1" alertBody="Test body1" />
    );
    expect(container).toMatchSnapshot();
  });

  it("should render a info alert", () => {
    const { container } = render(
      <SiteAlert slim icon={false} alertBody="Test body2" />
    );
    expect(container).toMatchSnapshot();
  });
});
