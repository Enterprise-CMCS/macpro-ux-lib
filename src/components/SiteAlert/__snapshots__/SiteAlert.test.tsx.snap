// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compontent snapshots should render a info alert 1`] = `
<div>
  <section
    aria-label="Site alert,"
    class="grid-row usa-site-alert usa-site-alert--info usa-site-alert--slim usa-site-alert--no-icon"
    role="alert"
  >
    <div
      class="usa-alert grid-col"
    >
      <div
        class="usa-alert__body"
      >
        <div
          class="usa-alert__text"
        >
          Test body2
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`compontent snapshots should render an emergency alert 1`] = `
<div>
  <section
    aria-label="Site alert,"
    class="grid-row usa-site-alert usa-site-alert--emergency"
    role="alert"
  >
    <div
      class="usa-alert grid-col"
    >
      <div
        class="usa-alert__body"
      >
        <h3
          class="usa-alert__heading"
        >
          Test1
        </h3>
        <div
          class="usa-alert__text"
        >
          Test body1
        </div>
      </div>
    </div>
  </section>
</div>
`;
